wikibaseintegrator/wbi_helpers.py

```diff
@@ -248,17 +253,17 @@ def execute_sparql_query(query: str, prefix: str = None, endpoint: str = None, u
         try:
             response = helpers_session.post(sparql_endpoint_url, params=params, headers=headers)
         except requests.exceptions.ConnectionError as e:
-            logging.error(f"Connection error: {e}. Sleeping for {retry_after} seconds.")
+            logging.error("Connection error: %s. Sleeping for %d seconds.", e, retry_after)
             sleep(retry_after)
             continue
         if response.status_code in (500, 502, 503, 504):
-            logging.error(f"Service unavailable (HTTP Code {response.status_code}). Sleeping for {retry_after} seconds.")
+            logging.error("Service unavailable (HTTP Code %d). Sleeping for %d seconds.", response.status_code, retry_after)
             sleep(retry_after)
             continue
         if response.status_code == 429:
             if 'retry-after' in response.headers.keys():
                 retry_after = int(response.headers['retry-after'])
-            logging.error(f"Too Many Requests (429). Sleeping for {retry_after} seconds")
+            logging.error("Too Many Requests (429). Sleeping for %d seconds", retry_after)

```

Comment:
Is there a reason for not using f-strings here?

Reply:
It's because of a performance issue with f-string, so some linter, like pylint, recommend using lazy formatting.
https://vald-phoenix.github.io/pylint-errors/plerr/errors/logging/W1202.html